server/Auth.js

```diff
@@ -21,12 +28,267 @@ class Auth {
+      }))
+    }
+
+    // Load the JwtStrategy (always) -> for bearer token auth 
+    passport.use(new JwtStrategy({
+      jwtFromRequest: ExtractJwt.fromExtractors([ExtractJwt.fromAuthHeaderAsBearerToken(), ExtractJwt.fromUrlQueryParameter('token')]),
+      secretOrKey: Database.serverSettings.tokenSecret
+    }, this.jwtAuthCheck.bind(this)))
+
+    // define how to seralize a user (to be put into the session)
+    passport.serializeUser(function (user, cb) {
+      process.nextTick(function () {
+        // only store id to session
+        return cb(null, JSON.stringify({
+          "id": user.id,
+        }))
+      })
+    })
+
+    // define how to deseralize a user (use the ID to get it from the database)
+    passport.deserializeUser((function (user, cb) {
+      process.nextTick((async function () {
+        const parsedUserInfo = JSON.parse(user)
+        // load the user by ID that is stored in the session
+        const dbUser = await Database.userModel.getUserById(parsedUserInfo.id)
+        return cb(null, dbUser)
+      }).bind(this))
+    }).bind(this))
+  }
+
+  /**
+   * Stores the client's choice how the login callback should happen in temp cookies
+   * 
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   */
+  paramsToCookies(req, res) {
+    if (req.query.isRest?.toLowerCase() == "true") {
+      // store the isRest flag to the is_rest cookie 
+      res.cookie('is_rest', req.query.isRest.toLowerCase(), {
+        maxAge: 120000, // 2 min
+        httpOnly: true
+      })
+    } else {
+      // no isRest-flag set -> set is_rest cookie to false
+      res.cookie('is_rest', "false", {
+        maxAge: 120000, // 2 min
+        httpOnly: true
+      })
+
+      // persist state if passed in
+      if (req.query.state) {
+        res.cookie('auth_state', req.query.state, {
+          maxAge: 120000, // 2 min
+          httpOnly: true
+        })
+      }
+
+      const callback = req.query.redirect_uri || req.query.callback
+
+      // check if we are missing a callback parameter - we need one if isRest=false
+      if (!callback) {
+        res.status(400).send({
+          message: 'No callback parameter'
+        })
+        return
+      }
+      // store the callback url to the auth_cb cookie 
+      res.cookie('auth_cb', callback, {
+        maxAge: 120000, // 2 min
+        httpOnly: true
+      })
+    }
+  }
+
+  /**
+   * Informs the client in the right mode about a successfull login and the token
+   * (clients choise is restored from cookies).
+   * 
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   */
+  async handleLoginSuccessBasedOnCookie(req, res) {
+    // get userLogin json (information about the user, server and the session)
+    const data_json = await this.getUserLoginResponsePayload(req.user)
+
+    if (req.cookies.is_rest === 'true') {
+      // REST request - send data
+      res.json(data_json)
```

Comment:
I think there should be a single comment what is_rest is exactly for as it confused me a bit during the app implementation. I used is_rest=true for the mobile app bc a redirect is not needed.